bin/reth/src/main.rs

```diff
@@ -91,32 +91,31 @@ fn main() {
                             builder.launch_with(launcher)
                         })
                         .await?;
-                    handle.node_exit_future.await
-                }
-                true => {
-                    info!(target: "reth::cli", "Running with legacy engine");
-                    let handle = builder.launch_node(EthereumNode::default()).await?;
+
                     let blockchain_provider = handle.node.provider.clone();
                     let config = handle.node.config.config.clone();
                     let chain = handle.node.chain_spec();
                     let datadir = handle.node.data_dir.clone();
-                    let (provider_factory, bitfinity) = handle.bitfinity_import.clone().expect("Bitfinity import not configured");
+                    let (provider_factory, bitfinity) = handle.bitfinity_import.clone().expect("Bitfinity import not configured");                    
```

Comment:
considering that line 74 is `let use_legacy_engine = true;` I am pretty sure this is never reached so the tool is broken.
Please start the node locally and check that it works as expected.

Reply:
Good point! Fixed and tested.